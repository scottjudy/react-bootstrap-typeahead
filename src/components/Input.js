// @flow

import cx from 'classnames';
import React from 'react';

type Props = {
  className?: string,
};

const Input = React.forwardRef<Props, ?HTMLInputElement>((props, ref) => (
  <input
    {...props}
    className={cx('rbt-input-main', props.className)}
    ref={ref}
  />
));

export default Input;
