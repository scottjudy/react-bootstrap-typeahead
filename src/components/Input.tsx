import cx from 'classnames';
import React, { forwardRef, HTMLProps } from 'react';

type InputProps = HTMLProps<HTMLInputElement>;

const Input = forwardRef<HTMLInputElement, InputProps>((props, ref) => (
  <input
    {...props}
    className={cx('rbt-input-main', props.className)}
    ref={ref}
  />
));

export default Input;
