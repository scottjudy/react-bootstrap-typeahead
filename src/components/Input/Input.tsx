import cx from 'classnames';
import React, { forwardRef, HTMLAttributes } from 'react';

export type InputProps = HTMLAttributes<HTMLInputElement>;

const Input = forwardRef<HTMLInputElement, InputProps>((props, ref) => (
  <input
    {...props}
    className={cx('rbt-input-main', props.className)}
    ref={ref}
  />
));

export default Input;
