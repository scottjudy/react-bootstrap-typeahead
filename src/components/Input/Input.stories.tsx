/* eslint-disable sort-keys,import/no-extraneous-dependencies */

import React from 'react';
import { Story, Meta } from '@storybook/react';

import Input, { InputProps } from './Input';

export default {
  title: 'Components/Input',
  component: Input,
} as Meta;

const Template: Story<InputProps> = (args) => <Input {...args} />;

export const Default = Template.bind({});
Default.args = {};
