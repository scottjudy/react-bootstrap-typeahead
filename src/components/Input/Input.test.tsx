import * as stories from './Input.stories';
import { generateSnapshots } from '../../tests/helpers';

describe('<Input>', () => {
  generateSnapshots(stories);
});
