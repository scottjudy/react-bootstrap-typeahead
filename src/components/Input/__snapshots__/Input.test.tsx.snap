// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> Default story renders snapshot 1`] = `
<input
  class="rbt-input-main"
/>
`;
